pkg/validation/utils.go

```diff
@@ -9,3 +16,45 @@ func prefixValues(prefix string, values ...string) []string {
 	}
 	return msgs
 }
+
+func IsEndpointAllowed(endpoint *url.URL, allowedDomains []string) bool {
```

Comment:
Please add a doc string for this function

Reply:
added